src/main/java/uk/gov/justice/digital/hmpps/cmd/api/model/DryRunNotification.kt

```diff
@@ -66,5 +80,71 @@ data class DryRunNotification(
         processed = false
       )
     }
+
+    fun LocalDateTime.getDateTimeFormattedForTemplate(): String {
+      val day = this.dayOfMonth
+      val ordinal = if (day in 11..13) {
+        "th"
+      } else when (day % 10) {
+        1 -> "st"
+        2 -> "nd"
+        3 -> "rd"
+        else -> "th"
+      }
+
+      return DateTimeFormatter.ofPattern("EEEE, d'$ordinal' MMMM").format(this)
+    }
+
+    private fun getNotificationType(shiftNotificationType: ShiftType, activity: String?): String {
+      return if (shiftNotificationType == ShiftType.SHIFT) {
+        if (activity == null) {
+          "shift"
+        } else {
+          "detail"
+        }
+      } else {
+        if (activity == null) {
+          "overtime shift"
+        } else {
+          "overtime detail"
+        }
+      }
+    }
+
+    private fun getOptionalTaskDescription(from: LocalDateTime, to: LocalDateTime, task: String?): String {
+      return if (!task.isNullOrEmpty()) {
+        if (from.toLocalTime().isAfter(LocalTime.MIN) && to.toLocalTime().isAfter(LocalTime.MIN)) {
+          "(${from.toLocalTime()} - ${to.toLocalTime()}) "
+        } else {
+          "(full day) "
+        }
+      } else ""
+    }
+
+    private fun getOptionalTaskTo(task: String?, communicationPreference: CommunicationPreference, shiftActionType: DetailModificationType): String {
+      return if (communicationPreference == CommunicationPreference.NONE && task != null && task.isNotEmpty()) {
```

Comment:
would be better with `isNullOrEmpty` rather than checking for `null` and then `isNotEmpty`